eget.js

```diff
@@ -0,0 +1,343 @@
+    }
+    if (options.tmpDir && typeof options.tmpDir !== 'string') {
+      throw new TypeError('tmpDir must be a string');
+    }
+    if (options.verbose && typeof options.verbose !== 'boolean') {
+      throw new TypeError('verbose must be a boolean');
+    }
+
+    /** @type {string} */
+    this.wasmPath = options.wasmPath ?? join(__dirname, "eget.wasm");
+
+    /** @type {string} */
+    this.tmpDir = options.tmpDir || "./tmp";
+
+    /** @type {boolean} */
+    this.verbose = options.verbose || false;
+  }
+
+  /**
+   * Logs a message if verbose mode is enabled.
+   * @param {string} message - Message to log
+   */
+  log(message) {
+    if (this.verbose) {
+      console.log(message);
+    }
+  }
+
+  /**
+   * Ensures a directory exists, creating it if necessary.
+   * @param {string} dirPath - Directory path to ensure
+   * @throws {Error} If directory cannot be created
+   */
+  async ensureDir(dirPath) {
+    try {
+      await mkdir(dirPath, { recursive: true });
+    } catch (error) {
+      if (error.code !== "EEXIST") throw error;
+    }
+  }
+
+  /**
+   * Downloads a file from a URL to a local path.
+   * @param {string} url - URL to download from
+   * @param {string} filePath - Local file path to save to
+   * @throws {Error} If download fails or network error occurs
+   */
+  async downloadFile(url, filePath) {
+    this.log(`Downloading: ${url}`);
+
+    try {
+      const response = await fetch(url);
+      if (!response.ok) {
+        throw new Error(`HTTP ${response.status}: ${response.statusText}`);
+      }
+
+      await this.ensureDir(dirname(filePath));
+      const arrayBuffer = await response.arrayBuffer();
+      const buffer = Buffer.from(arrayBuffer);
+      await writeFile(filePath, buffer);
+      this.log(`Saved to: ${filePath}`);
+    } catch (error) {
+      if (error.name === "TypeError" && error.message.includes("fetch")) {
+        throw new Error(`Network error downloading ${url}: ${error.message}`);
+      }
+      if (error.message.includes('HTTP')) {
+        throw error; // Re-throw HTTP errors as-is
+      }
+      throw new Error(`Failed to download ${url}: ${error.message}`);
+    }
+  }
+
+  /**
+   * Runs the eget WASM binary with the given arguments.
+   * @param {string[]} args - Command line arguments for eget
+   * @returns {Promise<RunResult>} Result of the operation
+   * @throws {Error} If WASM file cannot be loaded or unexpected error occurs
+   */
+  async run(args) {
+    let wasmBytes;
+    try {
+      wasmBytes = await readFile(this.wasmPath);
+    } catch (error) {
+      throw new Error(
+        `Failed to read WASM file ${this.wasmPath}: ${error.message}`
+      );
+    }
+
+    const wasi = new WASI({
+      version: "preview1",
+      args: ["eget.wasm", ...args],
+      env: process.env,
+      preopens: {
+        "/": process.cwd(),
+      },
+    });
+
+    let module, instance;
+    try {
+      module = await WebAssembly.compile(wasmBytes);
```

Comment:
I suspect you wanna cache module in a global and not re-read-recompile  every retry